### PR TITLE
feat(examples): add reproducible Evaluation + Optimization pipeline

### PR DESCRIPTION
**Pull request by Adonis-a233**
## Summary

Implements the reproducible Evaluation + Optimization closed loop for the issue
"设计并实现可复现的 Evaluation + Optimization pipeline".

Resolves #91

## What Changed

Adds `examples/optimization/eval_optimize_loop/` — a six-stage pipeline around
`AgentEvaluator` / `AgentOptimizer`:

1. **Baseline evaluation** — train and validation sets scored separately; per-case
   metric sub-scores (final_response / tool_trajectory / rubric), pass/fail,
   failure reasons, and key trace fields.
2. **Failure attribution** — rule-based clustering over structured trajectories into
   `final_response_mismatch` / `tool_call_error` / `parameter_error` /
   `llm_rubric_not_met` / `knowledge_recall_insufficient` / `format_error`.
   `case_meta.json` declares an expected category per case and the report carries an
   attribution accuracy **self-check** (4/4 = 100% on the bundled sample).
3. **Optimization** — fake mode applies a deterministic scripted candidate; live mode
   runs a real GEPA search via `AgentOptimizer.optimize` + `TargetPrompt.add_path`.
4. **Candidate validation** — full re-run and per-case diff vs baseline
   (`new_pass` / `new_fail` / `score_up` / `score_down`).
5. **Acceptance gate** — five independent configurable checks: validation gain
   threshold, no new hard fail, no key-case regression, no train-up/val-down
   overfit, cost within budget (optimizer spend + token-estimated evaluation spend).
6. **Audit persistence** — append-only `runs/<timestamp>_<run_id>/` per run with
   prompt snapshots, JSON/Markdown reports, gate reasons, cost/token split,
   duration, GEPA seed, prompt SHA-256, and a full config snapshot; `run_id` is
   injected into every log line for cross-artifact tracing.

Robustness and engineering:

- Live agent calls retry with exponential backoff + jitter and a per-call timeout
  (`EVAL_OPT_CALL_TIMEOUT` / `EVAL_OPT_CALL_ATTEMPTS` / `EVAL_OPT_CALL_BACKOFF`).
- `optimizer.json` is validated at startup (metric weights must sum to 1.0, all
  gate keys present) with readable errors instead of bare `KeyError`s.
- 33 IO-free unit tests under `tests/` cover attribution, rubric scoring, every
  gate check, case diffing, the self-check, and config validation.
- Generated reports are gitignored (running the example never dirties the tree);
  frozen samples are committed under `sample_output/` (`.json` + `.md`).
- The 6 sample cases cover all three required situations: optimizable success,
  optimization-ineffective, and post-optimization regression. README includes the
  requested design note (failure attribution, gating, overfit protection, audit).

## Validation

Fake mode (no API key, ~1s, deterministic):

```bash
python examples/optimization/eval_optimize_loop/run.py --mode fake

### CODE REVIEW

**Comment by Adonis-a233** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by raychen911** [comment]
Closing as related to #91.
